eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/CorrelationId.scala

```diff
@@ -0,0 +1,49 @@
+package com.evolutiongaming.kafka.journal.eventual.cassandra
+
+import cats.Monad
+import cats.effect.std.UUIDGen
+import cats.syntax.all.*
+import com.evolutiongaming.scassandra.*
+import com.datastax.driver.core.{GettableByNameData, SettableData}
+
+import scala.jdk.CollectionConverters.*
+
+final case class CorrelationId(value: String) extends AnyVal {
+  override def toString: String = value
+}
+
+object CorrelationId {
+  def unsafe(value: String): CorrelationId = new CorrelationId(value)
+
+  def of[F[_]: Monad: UUIDGen]: F[CorrelationId] =
+    UUIDGen[F]
+      .randomUUID
+      .map { uuid =>
+        CorrelationId(uuid.toString)
+      }
+
+  val key = "correlation_id"
+
+  implicit val encodeByNameCorrelationId: EncodeByName[CorrelationId] = EncodeByName[String].contramap(_.value)
+  implicit val decodeByNameCorrelationId: DecodeByName[CorrelationId] = DecodeByName[String].map(CorrelationId.unsafe)
+  implicit val encodeByIdxCorrelationId: EncodeByIdx[CorrelationId]   = EncodeByIdx[String].contramap(_.value)
+  implicit val decodeByIdxCorrelationId: DecodeByIdx[CorrelationId]   = DecodeByIdx[String].map(CorrelationId.unsafe)
+
+  implicit val encodeAsPropertiesCorrelationId: EncodeRow[Option[CorrelationId]] = new EncodeRow[Option[CorrelationId]] {
+    def apply[B <: SettableData[B]](data: B, value: Option[CorrelationId]) = {
+      value
+        .map {
+          case CorrelationId(value) =>
+            data.setMap("properties", Map(key -> value).asJava, classOf[String], classOf[String])
```

Comment:
IMHO we should work on way to *update*, not _reset_ the extra values... 

on related note: I was thinking about full data consistency checks, probably for that we will require some way to "mark" accessible data, like some GCs mark accessible objects with colored flags and remove all, which lack expected color. Maybe introducing extra column(s) is safer and easier way how to approach this problem, though adding columns for every feature is not a nice solution either...

Reply:
I've reimplemented logic of binding value on the statement: now its adding key-value to the map instead of overriding whole map. As for now the branch depends on locally build `scassandra`, PR still in review: https://github.com/evolution-gaming/scassandra/pull/263